cmd/alerts/list_alerts.go

```diff
@@ -0,0 +1,197 @@
+)
+
+var levelCmd string
+
+type logCapture struct {
+	buffer bytes.Buffer
+}
+
+type alertCmd struct {
+	clusterID string
+	getLevel  string
+	//active    bool
+}
+
+func (capture *logCapture) GetStdOut() string {
+	return capture.buffer.String()
+}
+
+func (capture *logCapture) Write(p []byte) (n int, err error) {
+	a := string(p)
+	capture.buffer.WriteString(a)
+	return len(p), nil
+}
+
+//osdctl alerts list ${CLUSTERID} --level [warning, critical, firing, pending, all] --active bool 
+
+func NewCmdListAlerts() *cobra.Command {
+	alertCmd := &alertCmd{}
+	newCmd := &cobra.Command{
+		Use:               "list <cluster-id>",
+		Short:             "list alerts",
+		Long:              `Checks the alerts for the cluster`,
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			alertCmd.clusterID = args[0]
+			ListCheck(alertCmd)
+		},
+	}
+
+	newCmd.Flags().StringVarP(&alertCmd.getLevel, "level", "", "", "Alert level [warning, critical, firing, pending, all]")
+	//newCmd.Flags().BoolVar(&alertCmd.active, "active", false, "Show only active alerts")
+
+	return newCmd
+}
+
+func getLevel(level string) string{
+	switch level{
+	case "warning":
+		levelCmd = "warning"
+	case "critical":
+		levelCmd = "critical"
+	case "firing":
+		levelCmd = "firing"
+	case "pending":
+		levelCmd = "pending"
+	case "all":
+		levelCmd = "all" 
+	default:
+		log.Fatalf("Invalid alert level: %s\n", level)
+		return ""
+	} 
+	return levelCmd
+}
+
+func ListCheck(cmd *alertCmd) {
+	clusterID := cmd.clusterID
+	levelCmd := cmd.getLevel
+	//active := cmd.active
+
+	defer func() {
+		if err := recover(); err != nil {
+			log.Fatal("error : ", err)
+		}
+	}()
+
+	_, kubeconfig, clientset, err := GetKubeCli(clusterID)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output, err := getAlerts(kubeconfig, clientset, LocalHostUrl, levelCmd, PodName)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Println("Print information from all alert", output)
+
+}
+
+func GetKubeCli(clusterID string) (client.Client, *rest.Config, *kubernetes.Clientset, error) {
+
+	scheme := runtime.NewScheme()
+	err := routev1.AddToScheme(scheme)
+	if err != nil {
+		fmt.Print("failed to register scheme")
+	}
+
+	bp, err := config.GetBackplaneConfiguration()
+	if err != nil {
+		log.Fatalf("failed to load backplane-cli config: %v", err)
+	}
+
+	kubeconfig, err := login.GetRestConfig(bp, clusterID)
+	if err != nil {
+		log.Fatalf("failed to load backplane admin: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		log.Fatalf("failed to create clientset : %v", err)
+	}
+
+	kubeCli, err := client.New(kubeconfig, client.Options{})
+	if err != nil {
+		log.Fatalf("failed to load kubecli : %v", err)
+	}
+
+	return kubeCli, kubeconfig, clientset, err
+}
+
+func getAlerts(kubeconfig *rest.Config, clientset *kubernetes.Clientset, LocalHostUrl string, levelCmd string, PodName string) (string, error) {
```

Comment:
`LocalHostUrl` is incorrectly capitalized as an argument, but it's also a `const` in this file.

Unless it's needed for a test you probably don't have to pass the `const` in explicitly.

`PodName` is also incorrectly capitalized & a const.

So these would be shadowing the const vars (if the values can be different).

Reply:
@bergmannf Created `utility` inside `alerts` https://github.com/openshift/osdctl/pull/500/commits/2c49c094ad8077e3bf0bce597ba16f3dab524470#diff-75037fd13be730126f5b1e01769f6f07fb98f9e53bc4fbb2622b295d6ea7a933 
`LocalHostUrl` and `PodName` declared as var